apps/dbagent/src/app/api/mcp/servers/upload/route.ts

```
@@ -0,0 +1,50 @@
+import { exec } from 'child_process';
+import { promises as fs } from 'fs';
+import { NextResponse } from 'next/server';
+import path from 'path';
+import { promisify } from 'util';
+
+const execAsync = promisify(exec);
+
+export async function POST(request: Request) {
+  try {
+    const formData = await request.formData();
+    const file = formData.get('file') as File;
+
+    if (!file) {
+      return NextResponse.json({ message: 'No file provided' }, { status: 400 });
+    }
+
+    if (!file.name.endsWith('.ts')) {
+      return NextResponse.json({ message: 'Only TypeScript (.ts) files are allowed' }, { status: 400 });
+    }
+
+    // Ensure the mcp-source directory exists
+    const mcpSourceDir = path.join(process.cwd(), 'mcp-source');
+    try {
+      await fs.access(mcpSourceDir);
+    } catch {
+      await fs.mkdir(mcpSourceDir, { recursive: true });
```

I'd prefer not creating the directory when it doesn't exist, but rather raise an error.

---

Yeah, I was thinking in case the directory is deleted, but that could cause some issues down the line with creating something new. I will change it to add an error instead.